instrumentation/netty/netty-4-common/library/src/main/java/io/opentelemetry/instrumentation/netty/v4/common/internal/client/NettyClientInstrumenterFactory.java

```diff
@@ -19,20 +19,36 @@
 import io.opentelemetry.instrumentation.api.instrumenter.net.PeerServiceAttributesExtractor;
 import io.opentelemetry.instrumentation.netty.common.internal.HttpClientSpanKeyAttributesExtractor;
 import io.opentelemetry.instrumentation.netty.common.internal.NettyConnectionRequest;
-import io.opentelemetry.javaagent.bootstrap.internal.CommonConfig;
-import io.opentelemetry.javaagent.instrumentation.netty.v4.common.HttpRequestAndChannel;
+import io.opentelemetry.instrumentation.netty.v4.common.internal.HttpRequestAndChannel;
+import java.util.List;
+import java.util.Map;
 
+/**
+ * This class is internal and is hence not for public use. Its APIs are unstable and can change at
+ * any time.
+ */
 public final class NettyClientInstrumenterFactory {
 
   private final String instrumentationName;
   private final boolean connectionTelemetryEnabled;
   private final boolean sslTelemetryEnabled;
+  private final List<String> capturedRequestHeaders;
+  private final List<String> capturedResponseHeaders;
+  private final Map<String, String> peerServiceMapping;
 
   public NettyClientInstrumenterFactory(
-      String instrumentationName, boolean connectionTelemetryEnabled, boolean sslTelemetryEnabled) {
+      String instrumentationName,
+      boolean connectionTelemetryEnabled,
+      boolean sslTelemetryEnabled,
+      List<String> capturedRequestHeaders,
+      List<String> capturedResponseHeaders,
+      Map<String, String> peerServiceMapping) {
```

Comment:
This is better suited as a builder, but I'd like to wait until I break out `:instrumentation:netty:netty-4.1` which will really demonstrate the merits of using a builder because these will be optional fields for the user.